### ISSUE TITLE
RefCOCO/+/g results: 0.6% lower than reported

### ISSUE THREAD

**Issue by JiaHeng-DLUT** [om-ai-lab/VLM-R1#68]
I test the [omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps](https://huggingface.co/omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps) model on the RefCOCO/+/g datasets. My results:

| RefCOCO | | | RefCOCO+ | | | RefCOCOg | | Average |
|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|
| val | test_A | test_B | val | test_A | test_B | val | test | val sets |
| 90.6 | 92.8 | 85.8 | 83.8 | 89.4 | 77.6 | 84.8 | 86.4 | 86.4 |

The average across all val sets is 86.4, which is 0.6 lower than your reported figure. 
Is this difference within expected variance, or might I be missing something in my evaluation setup?

**Comment by StinkyTofu95** [comment]
> I test the [omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps](https://huggingface.co/omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps) model on the RefCOCO/+/g datasets. My results:
> 
> RefCOCO			RefCOCO+			RefCOCOg		Average
> val	test_A	test_B	val	test_A	test_B	val	test	val sets
> 90.6	92.8	85.8	83.8	89.4	77.6	84.8	86.4	86.4
> The average across all val sets is 86.4, which is 0.6 lower than your reported figure. Is this difference within expected variance, or might I be missing something in my evaluation setup?

I think there is some randomness in GRPO training itself? After all, there will be N randomly sampled outputs.

**Comment by JiaHeng-DLUT** [comment]
> > I test the [omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps](https://huggingface.co/omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps) model on the RefCOCO/+/g datasets. My results:
> > RefCOCO			RefCOCO+			RefCOCOg		Average
> > val	test_A	test_B	val	test_A	test_B	val	test	val sets
> > 90.6	92.8	85.8	83.8	89.4	77.6	84.8	86.4	86.4
> > The average across all val sets is 86.4, which is 0.6 lower than your reported figure. Is this difference within expected variance, or might I be missing something in my evaluation setup?
> 
> I think there is some randomness in GRPO training itself? After all, there will be N randomly sampled outputs.

I did not train the model myself - I simply tested the authors' pre-trained model and got 86.4, which is 0.6 lower than their reported figure.

**Comment by SZhanZ** [comment]
Hello , this may be due to runtime errors caused by different machines?

**Comment by JiaHeng-DLUT** [comment]
Thanks for your response. I also test [Qwen/Qwen2.5-VL-3B-Instruct](https://huggingface.co/Qwen/Qwen2.5-VL-3B-Instruct).

|                            Model                             |                         Data source                          | RefCOCO |        |        | RefCOCO+ |        |        | RefCOCOg |      | RefGTA | RefCOCO val ave. |
| :----------------------------------------------------------: | :----------------------------------------------------------: | :-----: | :----: | :----: | :------: | :----: | :----: | :------: | :--: | :----: | :--------------: |
|                                                              |                                                              |   val   | test_A | test_B |   val    | test_A | test_B |   val    | test |   -    |                  |
| [Qwen/Qwen2.5-VL-3B-Instruct](https://github.com/om-ai-lab/VLM-R1) | Figure and [issue 6](https://github.com/om-ai-lab/VLM-R1/issues/6#issuecomment-2667470902) |  77.8   |   -    |   -    |   71.6   |   -    |   -    |   71.8   |  -   |   -    |       73.7       |
|                                                              |                           Our test                           |  77.8   |  85.0  |  73.0  |   72.8   |  82.0  |  67.8  |   73.2   | 73.4 |  71.8  |       74.6       |
| [omlab/Qwen2.5VL-3B-VLM-R1-REC-500steps](https://github.com/om-ai-lab/VLM-R1) |                            Figure                            |    -    |   -    |   -    |    -     |   -    |   -    |    -     |  -   |  74.6  |       87.0       |
|                                                              |                           Our test                           |  90.6   |  92.8  |  85.8  |   83.8   |  89.4  |  77.6  |   84.8   | 86.4 |  74.6  |       86.4       |

In summary, the base model increased slightly (73.7→74.6), while the R1 version dropped a bit (87.0→86.4).

**Comment by JiaHeng-DLUT** [comment]
> Hello , this may be due to runtime errors caused by different machines?

Do you evaluate on the entire test set or only the first 500 samples? Just want to clarify your testing protocol.
